src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -29,34 +28,27 @@ import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import type SCREENS from '@src/SCREENS';
-import type {LoginList, Policy, SecurityGroup, Session as TSession} from '@src/types/onyx';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
 import ValidateCodeForm from './ValidateCodeForm';
 import type {ValidateCodeFormHandle} from './ValidateCodeForm/BaseValidateCodeForm';
 
-type ContactMethodDetailsPageOnyxProps = {
-    /** Login list for the user that is signed in */
-    loginList: OnyxEntry<LoginList>;
+type ContactMethodDetailsPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.PROFILE.CONTACT_METHOD_DETAILS>;
 
-    /** Current user session */
-    session: OnyxEntry<TSession>;
-
-    /** User's security group IDs by domain */
-    myDomainSecurityGroups: OnyxEntry<Record<string, string>>;
-
-    /** All of the user's security groups and their settings */
-    securityGroups: OnyxCollection<SecurityGroup>;
-
-    /** Indicated whether the report data is loading */
-    isLoadingReportData: OnyxEntry<boolean>;
-
-    /** The list of this user's policies */
-    policies: OnyxCollection<Pick<Policy, 'id' | 'ownerAccountID' | 'owner'>>;
-};
-
-type ContactMethodDetailsPageProps = ContactMethodDetailsPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.PROFILE.CONTACT_METHOD_DETAILS>;
+function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
+    const [loginList, loginListResult] = useOnyx(ONYXKEYS.LOGIN_LIST);
+    const [session, sessionResult] = useOnyx(ONYXKEYS.SESSION);
+    const [myDomainSecurityGroups, myDomainSecurityGroupsResult] = useOnyx(ONYXKEYS.MY_DOMAIN_SECURITY_GROUPS);
+    const [securityGroups, securityGroupsResult] = useOnyx(ONYXKEYS.COLLECTION.SECURITY_GROUP);
+    const [isLoadingReportData, isLoadingReportDataResult] = useOnyx(ONYXKEYS.IS_LOADING_REPORT_DATA, {initialValue: true});
+    const [policies, policiesResult] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {
+        selector: (data) => ({
```

Comment:
How about a rule to always define selectors as separate function? And reuse exactly like with `chatReportSelector` function

Reply:
I will extract to a separate function but we shouldn't take this as a rule, because I would want to use some component's data inside that selector, e.g.:

```tsx
    const [isLoadingReportData, isLoadingReportDataResult] = useOnyx(ONYXKEYS.IS_LOADING_REPORT_DATA, {initialValue: true});
    const [policies, policiesResult] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {
        selector: (data) => ({
            id: isLoadingReportData ? data?.id ?? '' : 'something', // <- using external data inside selector
            ownerAccountID: data?.ownerAccountID,
            owner: data?.owner ?? '',
        }),
    });
```